File: AlipayTest/ViewController.m
//
//  ViewController.m
//  AlipayTest
//
//  Created by FOODING on 16/4/22.
//  Copyright © 2016年 FOODING. All rights reserved.
//

#import "ViewController.h"
#import "AppDelegate.h"

#import "SecondViewController.h"

@interface ViewController ()
{
    
}



@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    AppDelegate *app = KAPPDELEGATE;
    
     
    __weak ViewController *weakSelf = self;
    
    _aliView = [[AliView alloc] initWithFrame:CGRectMake(0, 100, SCREEN_WIDTH, SCREEN_HEIGHT - 200) withHasMore:YES];
    
    [_aliView loadBtnArrWithArr:app.currentArray];
    
    
    
    _aliView.moreBlock = ^
    {
        SecondViewController *secondVC = [[SecondViewController alloc] init];
        
        [weakSelf.navigationController showViewController:secondVC sender:nil];
        
        secondVC.popBlock = ^(NSArray *arr)
        {
            [weakSelf.aliView reloadBtnArrWithArr:arr];
            
        };
        
    };
    
    _aliView.clickBlock = ^(Model *aModel)
    {
        NSLog(@"当前点击的是%@", aModel.title);
    };
    
    [self.view addSubview:_aliView];
    
    
    
}



- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: AlipaySwift/ViewController.swift
//
//  ViewController.swift
//  AlipaySwift
//
//  Created by FOODING on 17/1/41.
//  Copyright © 2017年 Noohle. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    var gestureView = UIView()
    
    var startPoint = CGPoint.zero
    
    let width = Int(UIScreen.main.bounds.size.width / 4)
    
    let height = 80
    
    var aliview = AliView() 
    

    override func viewDidLoad() {
        super.viewDidLoad()
        

        
        aliview = AliView.init(frame: CGRect(x: 0, y: 100, width: width * 4, height: height * 3), hasMore: true)
        
        aliview.backgroundColor = UIColor.cyan
        aliview.loadModels(arr: Util.currentData)
        
        aliview.moreBlock = {
            self.goToNext()
        }
        
        aliview.modelBlock = {
            model in
            print(model.title)
        }
        
        self.view.addSubview(aliview)
        
    }
    
    

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    func goToNext() {
        
        let main = UIStoryboard(name: "Main", bundle: nil)
        
        let next = main.instantiateViewController(withIdentifier: "next") as! NextViewController
        
        next.popBlock = {
            data in 
            
            self.aliview.reloadModels(arr: data)
        }
        
        self.navigationController?.pushViewController(next, animated: true)

    }
    
    


}

